### PR TITLE
add epsilon maps

### PR DESCRIPTION
**Pull request by olliethomas**
This isn't that clever. Epsilon has gone from being a single value to a map of col_id number to float. 
At the same time, we no longer need exactly 4 epsilons for plots, we just do one experiment at a time.
Additionally, instead of `plt.show()`-ing the plots, we save them in the hydra outputs dir for that run (which has the config etc stored as well)

### CODE REVIEW

File: src/dynamics/response.py [hunk 1]
```diff
@@ -35,13 +35,15 @@ def respond(
...
+        epsilon_map = unwrap_or(self.epsilon, default={a: 1 for a in list(changeable_features)})
         new_features = np.copy(features)
 
         neg_item_mask = np.dot(action[None, :], features[...].T)[0] < 0
         assert not isinstance(changeable_features, slice)
         for changeable_feature in changeable_features:
+            weight = epsilon_map[changeable_feature]
             new_features[np.nonzero(neg_item_mask), changeable_feature] += (
-                self.epsilon / np.linalg.norm(action)
+                weight / np.linalg.norm(action)
```
Comment (olliethomas) [inline src/dynamics/response.py:44]: I'm not sure on this line

Comment (olliethomas) [reply src/dynamics/response.py:44]: I guess the question is how should epsilon for this value be interpreted: as an absolute, or relative distance

**Comment by olliethomas** [comment]
I wouldn't be against getting rid of the changeable features input and just controlling what features are changeable via the epsilon map.

File: src/dynamics/response.py (hunk 1 above)
Comment (tmke8) [reply src/dynamics/response.py:44]: Hmm, I'm not sure I understand. The way you did it here seems like the most straight-forward way to do it.

Comment (olliethomas) [reply src/dynamics/response.py:44]: 😅 I mean that I'm not sure about dividing by the norm of the entire weight vector, and then I was mostly talking to myself about the merits of just deleting it

**Comment by tmke8** [comment]
> I wouldn't be against getting rid of the changeable features input and just controlling what features are changeable via the epsilon map.

Yes, that sounds good. The unchangeable features can just have epsilon set to 0, right?

Btw, am I understanding this PR right that it removes the multiplots which compared different epsilon settings?

EDIT: ah sorry, you explained this in the PR description. hmm, it does make sense that we don't always want to run exactly 4 experiments. so, sure, let's just save the plots

File: src/dynamics/response.py (hunk 1 above)
Comment (tmke8) [reply src/dynamics/response.py:44]: ah, that part. yeah, one reason to delete it is that Perdomo et al. don't have it (I think?)

**Review by tmke8 — APPROVED** [review]
LGTM, though it would probably be nicer if the "actionable features" mechanism was removed, as you mentioned

File: src/dynamics/response.py (hunk 1 above)
Comment (olliethomas) [reply src/dynamics/response.py:44]: Leaving it for now, but have put a TODO comment  here for discussion at a later date

**Pull request merged**